src/nwb_conversion_tools/nwbconverter.py

```diff
@@ -172,28 +171,21 @@ def run_conversion(
         self.validate_conversion_options(conversion_options=conversion_options_to_run)
 
         if save_to_file:
-            load_kwargs = dict(path=nwbfile_path)
-            if nwbfile_path is None:
-                raise TypeError("A path to the output file must be provided, but nwbfile_path got value None")
-            if Path(nwbfile_path).is_file() and not overwrite:
-                load_kwargs.update(mode="r+", load_namespaces=True)
+            load_kwargs = dict(nwbfile_path=nwbfile_path, overwrite=overwrite)
+            if nwbfile is None or (not Path(nwbfile_path).exists() and not overwrite) or overwrite:
+                load_kwargs.update(metadata=metadata)
             else:
-                load_kwargs.update(mode="w")
-            with NWBHDF5IO(**load_kwargs) as io:
-                if load_kwargs["mode"] == "r+":
-                    nwbfile = io.read()
-                elif nwbfile is None:
-                    nwbfile = make_nwbfile_from_metadata(metadata=metadata)
+                load_kwargs.update(nwbfile=nwbfile)
+            with make_or_load_nwbfile(**load_kwargs) as nwbfile_out:
```

Comment:
This rather convoluted block is done in order to prevent the warnings for the input arguments: https://github.com/catalystneuro/nwb-conversion-tools/blob/main/src/nwb_conversion_tools/tools/nwb_helpers.py#L142-L150

I can think of two alternatives:

(i) removing those warnings completely

(ii) adding verbosity control to suppress them and allow us to just pass all 4 inputs to the function.

Reply:
@h-mayorquin which do you think is preferable?

Reply:
I think removing the warnings is better.